File: recursive_addition/c_recursive_addition/recursive_addition.c
#include <stdio.h>
#include <stdlib.h>

int rec_add( int, int );

int main( int argc, char *argv[] ) {
  if( argc != 1 ) {
    fprintf( stderr, "Usage: %s <noargs>\n", argv[0]);
    exit(1);
  }

  printf( "add: %d\n", rec_add(6, 8));
  printf( "add: %d\n", rec_add(9, 14));
  return 0;
}

int rec_add( int x, int y ) {
  if( y == 0 ) {
    return x;
  } else {
    return 1 + rec_add( x, y - 1 );
  }
}

File: recursive_addition/golang_recursive_addition/recursive_addition.go
package main

import "os"
import "fmt"

func rec_add(  x int,  y int ) int {
  if( y == 0 ) {
    return x;
  } else {
    return 1 + rec_add( x, y - 1 );
  }
}

func main() {
  args := os.Args
  argc := len(args)

  if argc != 1 {
    fmt.Println(args[0] + " <noargs>")
    os.Exit(1)
  }

  fmt.Println("example 6+8=", rec_add(6,8))
}
